includes/wikia/api/CrossWikiArticlesApiController.class.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Controller to fetch information about articles across multiple
+ * wikis. Used by recommendations service to avoid multiple calls
+ * to MW.
+ *
+ * @author Mariusz Strzelecki <mstrzelecki@fandom.com>
+ */
+
+class CrossWikiArticlesApiController extends WikiaApiController {
+
+	/**
+	 * Get details about one or more articles
+	 *
+	 * @requestParam string $ids A string with a comma-separated list of pairs wikiID_articleId
+	 *
+	 * @responseParam array $items A list of results with the article ID as the index, each item has a title, url and thumbnail url
+	 *
+	 * @example $ids=177996_68580,2233_1678
+	 */
+	public function getDetails() {
+		$this->setOutputFieldType( "items", self::OUTPUT_FIELD_TYPE_OBJECT );
+                $articles = explode( ',', $this->request->getVal( 'ids', null ) );
+
+		// mapping parameters to wikiId => listOfArticles, and preparing cross-wiki sql
+		$wikiToArticleMap = array();
+		$sqlFilters = array();
+		foreach ( $articles as $article ) {
+			list( $wikiId, $articleId ) = explode('_', $article);
+			if ( ! isset($wikiToArticleMap[$wikiId]) ) {
+				$wikiToArticleMap[$wikiId] = array();
+			}
+			$wikiToArticleMap[$wikiId][] = $articleId;
+			$sqlFilters[] = sprintf('(page_wikia_id=%s and page_id=%s)', $wikiId, $articleId);
```

Comment:
Let's cast IDs here to integer to avoid SQL injection,

Reply:
Thanks! I'm ashamed for introducing this SQL injection here 🙈

Reply:
No worries :)